lib/utils/simnet-connext-channels.ts

```diff
@@ -0,0 +1,196 @@
+        channelBalance: balanceMap[1].channelBalance,
+      };
+    }),
+    mergeMap((balances) => {
+      // if the walletBalance is below the minimum threshold we error in
+      // order to trigger the faucet request
+      if (balances.walletBalance < minimumWalletBalance) {
+        return throwError(`balance under the threshold of ${minimumWalletBalance}`);
+      } else {
+        return of(balances);
+      }
+    }),
+    catchError(() => {
+      // the balance is under our specified threshold
+      // we'll hit the faucet with our connext address
+      // and then recheck the balance
+      return getConnextAddressObservable(client).pipe(
+        mergeMap((connextAddress) => {
+          return from(faucetRequest(connextAddress)).pipe(
+            delay(3000), // we wait 3 seconds before checking the balance again
+            mergeMap(() => checkBalanceObservable(client, currency, minimumWalletBalance)),
+          );
+        }),
+      );
+    }),
+  );
+};
+
+const getInfo = async (client: XudClient): Promise<GetInfoResponse.AsObject> => {
+  const request = new GetInfoRequest();
+  const info = await new Promise((resolve, reject) => {
+    client.getInfo(request, processResponse(resolve, reject));
+  });
+  return info as GetInfoResponse.AsObject;
+};
+
+const getConnextAddressObservable = (client: XudClient): Observable<string> => {
+  return from(getInfo(client)).pipe(
+    mergeMap((info: GetInfoResponse.AsObject) => {
+      if (info.connext && info.connext.address) {
+        return of(info.connext.address);
+      }
+      return throwError('connext address not found');
+    }),
+    catchError((_error, caught) => {
+      return caught.pipe(
+        delay(5000), // we wait 5 seconds before trying getinfo again
+      );
+    }),
+  );
+};
+
+const faucetRequest = (connextAddress: string) => {
+  return new Promise((resolve, reject) => {
+    const options: http.RequestOptions = {
+      method: 'POST',
+      hostname: '35.193.71.174',
+      port: 9000,
+      path: '/faucet',
+    };
+
+    const payload = {
+      address: connextAddress,
+    };
+
+    const payloadStr = JSON.stringify(payload);
+    options.headers = {
+      'Content-Type': 'application/json',
+      'Content-Length': Buffer.byteLength(payloadStr),
+    };
+
+    const req = http.request(options, (res) => {
+      switch (res.statusCode) {
+        case 200:
+          resolve(res);
+          break;
+        default:
+          reject('faucet request failed');
+          break;
+      }
+    });
+
+    req.on('error', reject);
+
+    if (payloadStr) {
+      req.write(payloadStr);
+    }
+    req.end();
+  });
+};
+
+const createSimnetChannel = (
+  { currency, minChannelAmount, minWalletAmount, retryInterval }:
+  { currency: string, minChannelAmount: number, minWalletAmount: number, retryInterval: number },
+) => {
+  const client$ = from(loadXudClient({}));
+  const balances$ = client$.pipe(
```

Comment:
Is this line (along with the one above) equivalent to doing 

```
const client = await loadXudClient();
const balances$ = checkBalanceObservable(client, currency, minWalletAmount);
```

The `await` approach seems more readable to me but maybe I'm just not used to rxjs enough or whether it's ok to mix & match `await` statements like this. Or maybe the downside would be having to make `createSimnetChannel` async, and this is just analagous to using promises `then` function?

Reply:
> The await approach seems more readable to me but maybe I'm just not used to rxjs enough or whether it's ok to mix & match await statements like this.

I agree it looks cleaner and you certainly can convert static values to observables that will next and complete immediately.

````
const client = await loadXudClient();
of(client) // creates a static observable to be mixed with other observables
```

>  the downside would be having to make createSimnetChannel async

That was mainly the reason, yes. In addition it's considered a good practice to not use higher order function's variables for inputs. It is recommended to use the stream's input as an output for the inner observables. That makes it easier to extract the observables into smaller functions down the road. I'm partially guilty of ignoring some of these in this PR.

I'll leave it as is for now.